### ISSUE TITLE
incorrect determinant computation in ring with inverses

### ISSUE THREAD

**Issue by pzinn** [Macaulay2/M2#2346]
this is yet another bug in rings with `Inverses=>true`:
```
i1 : k=4;

i2 : R=ZZ[u_0..u_(k-1),Inverses=>true,MonomialOrder=>Lex];

i3 : matrix table(k,k,(i,j)->u_i^j)

o3 = | 1 u_0 u_0^2 u_0^3 |
     | 1 u_1 u_1^2 u_1^3 |
     | 1 u_2 u_2^2 u_2^3 |
     | 1 u_3 u_3^2 u_3^3 |

             4       4
o3 : Matrix R  <--- R

i4 : det oo

o4 = 0

o4 : R
```

**Comment by mikestillman** [comment]
rats...  Assigning myself!

**Comment by mikestillman** [comment]
@DanGrayson This is another case where Bareiss method fails, but is being chosen as the default.

**Comment by DanGrayson** [comment]
Let's fix that, too.  Is the problem that there are inverses in the ring?

**Comment by mikestillman** [comment]
@DanGrayson There are a number of inverse crashes/problems occuring (i.e. several github issues).  Basically all the same problem, I think.  I'm working on them.

**Comment by mikestillman** [comment]
Fixed in PR #2480.
